pkg/image/api/types.go

```diff
@@ -106,6 +128,32 @@ type TagEvent struct {
 	DockerImageReference string
 	// The image
 	Image string
+	// Generation is the spec tag generation that resulted in this tag being updated
+	Generation int64
+}
+
+type TagEventConditionType string
+
+// These are valid conditions of TagEvents.
+const (
+	// ImportSuccess with status False means the import of the specific tag failed
+	ImportSuccess TagEventConditionType = "ImportSuccess"
+)
+
+// TagEventCondition contains condition information for a tag event.
+type TagEventCondition struct {
+	// Type of tag event condition, currently only ImportSuccess
+	Type TagEventConditionType
+	// Status of the condition, one of True, False, Unknown.
+	Status kapi.ConditionStatus
+	// Last time the condition transit from one status to another.
+	LastTransitionTime unversioned.Time
+	// (brief) reason for the condition's last transition.
+	Reason string
+	// Human readable message indicating details about last transition.
+	Message string
+	// Generation is the spec tag generation that this status corresponds to
+	Generation int64
```

Comment:
If the TagReference can have this as nil, why can't the TagEventCondition also have this as nil?